### DISCUSSION TITLE
how do i detect request hostname inside getServerSideProps ?

### DISCUSSION THREAD

**Discussion by meisamhasani** [vercel/next.js#71026] [Help]
### Summary

hello
how do i detect request hostname inside getServerSideProps ?
i need call fetch with same hostname with browser per request , and then render server side page
async function getServerSideProps() {
var res = await fetch("https://**[request.host]**/client/cms/Articles?take=3",

**Comment by suprunchuk** [comment]
To detect the request hostname inside `getServerSideProps`, you can use the `context` parameter that Next.js provides. The `context` object contains information about the incoming request, including the `host` header. Here's how you can modify your `getServerSideProps` function to use the request hostname:

```tsx
import { GetServerSideProps } from 'next'

export const getServerSideProps: GetServerSideProps = async (context) => {
  const { req } = context
  const protocol = process.env.NODE_ENV === 'production' ? 'https' : 'http'
  const host = req.headers.host || 'localhost'

  const url = `${protocol}://${host}/client/cms/Articles?take=3`
  
  const res = await fetch(url)
  const data = await res.json()

  return {
    props: {
      articles: data,
    },
  }
}
```

**Comment by meisamhasani** [comment]
thank you very match

**Comment by meisamhasani** [comment]
i have this error now : 
"getServerSideProps" is not supported in app/
my location is : /app/articles/[id]/page.tsx
I understand from the host what merchant the request is for. So that I can recover the data of that merchant
@suprunchuk
